request_information.go

```diff
@@ -31,6 +31,8 @@ type RequestInformation struct {
 	// The Url template for the current request.
 	UrlTemplate string
 	options     map[string]RequestOption
+	// Response handler to use in place of the default response handling provided by the core service
+	ResponseHandler ResponseHandler
```

Comment:
revert, instead add a RequestHandlerOption interface (inherits from RequestOption, has a getter and a setter to a request handler). And add a struct that implements it. People will be able to provide that "callback" through the request options (via request configuration) this way.